src/network/networkd-dhcp-common.c

```diff
@@ -61,7 +61,14 @@ void network_adjust_dhcp(Network *network) {
 DUID* link_get_duid(Link *link) {
         if (link->network->duid.type != _DUID_TYPE_INVALID)
                 return &link->network->duid;
-        else
+        else if (link->hw_addr.length == 0
+                 && (link->manager->duid.type == DUID_TYPE_LLT ||
+                     link->manager->duid.type == DUID_TYPE_LL)) {
+                link->network->duid.type = DUID_TYPE_EN;
+                // Fallback to DUID that works without mac addresses.
+                // This is useful for tunnel devices without mac address.
+                return &link->network->duid;
+        } else
```

Comment:
Please do not change values in `Network` object, as .network file may be assigned to multiple interfaces.

Reply:
fixed.

Reply:
> Please do not change values in `Network` object, as .network file may be assigned to multiple interfaces.

This is interesting because `link_configure_duid` mutates duid as well with link specific data. Is this broken?